src/oop_ext/foundation/immutable.py

```diff
@@ -21,10 +21,12 @@ def RegisterAsImmutable(immutable_type):
     # Fix it for the copy too!
     import copy
 
-    copy._copy_dispatch[immutable_type] = copy._copy_immutable
+    copy._copy_dispatch[  # type:ignore[attr-defined]
+        immutable_type
+    ] = copy._copy_immutable  # type:ignore[attr-defined]
 
 
-def AsImmutable(value, return_str_if_not_expected=True):
+def AsImmutable(value: Any, return_str_if_not_expected: bool = True) -> Any:
```

Comment:
How about using an overload here? 
If it's too much for this PR we could open an issue to tackle this.

Reply:
hmm seems that my ideia it's not possible to implement =/ 
I was thinking something like this
```
T = TypeVar('T', bound=(int, float, str, bool))

@overload
def AsImmutable(value: Union[List, Tuple] ...) -> Tuple:

@overload
def AsImmutable(value: Dict...) -> ImmutableDict:
```

But I missed the last line where any other object should return `str` =/